### ISSUE TITLE
[BUG] 1inch-LOP introduces duplicates into dex.trades

### ISSUE THREAD

**Issue by 0xBoxer** [duneanalytics/spellbook#7163]
### Description
1inch-LOP models are introducing duplicates into dex.trades. Swap events that belong to other protocols are decoded as being 1inch LOP swaps with an incorrect evt_index number.

Example queries: 

single example: https://dune.com/queries/4306571
multiple examples: https://dune.com/queries/4306524

### Expected behavior
1inch-LOP model should not pick up these trades 

### Impacted model(s)

```
dex.trades
oneinch.lop
```

### Possible solution
turns out that articifcially creating the event_index here is the problem

https://github.com/duneanalytics/spellbook/blob/90c8c376ee62a7c0ef187b88fdf55e4da79a601c/dbt_subprojects/dex/models/_projects/oneinch/oneinch_lop_own_trades.sql#L39

**Comment by jeff-dude** [comment]
@grkhr @max-morrow what do you think of this issue?

**Comment by grkhr** [comment]
These are not duplicates. In the first example everything is correct as in this TX a lot of 1inch limit orders have been filled (=our own liquidity). 

The issue is in `evt_index` logic. We don't emit trade params in `OrderFilled` event, so we take them from traces, and there is no way to properly match subtrace with event index (e.g. join the trade from first example with the corresponding event), so we do `row_number() over(partition by tx_hash order by call_trace_address) as evt_index` to align with table's unique key.

In my honest opinion, `evt_index` shouldn't present in `dex.trades` at all, as it's assuming that the dex project will emit trade event, but generally speaking it doesn't have to. But in current architecture it's all we can do with `evt_index`.

**Comment by 0xBoxer** [comment]
fair assessment, adding trace_address seems logically consistent to me. 

@jeff-dude I guess to not run into the `null <> null` problem we will need to deploy a surrogate key that combines a bunch of different factors together and can't be null

**Comment by jeff-dude** [comment]
closing issue due to inactivity. if still relevant, and proposed fix is available, feel free to reopen an issue as needed.

**Comment by 0xBoxer** [comment]
@jeff-dude I disagree with this approach, let's keep known data issues open, we will just loose track of them if we close the gh issues

**Comment by arijitAD** [comment]
@0xBoxer Are you planning to fix this issue?

**Comment by mrAndreyIsachenko** [comment]
This looks fixed already - #9747 (merged 2026-06-11, CUR2-2693) addresses the double counting directly: when a 1inch-LOP resolver settles by routing through an underlying DEX in the same tx, both the '1inch-LOP' row and the venue's own row used to land in dex.trades for the same fill. That PR adds oneinch_<chain>_lop_venue_settled_fills to detect and exclude those fills from oneinch_lop_own_trades (reclassifying them into dex_aggregator.trades instead), with a 6h maturity delay to avoid a raw-traces-vs-decoded-events race.

Checked prod today (2026-08-15): oneinch_ethereum.lop_venue_settled_fills has 9,961 rows for this month alone, so the exclusion is actively running, not just merged and idle.

Worth noting the root cause turned out to be a bit different from the evt_index/row_number() theory earlier in this thread - that numbering is still there, just now computed before the venue-settled filter so kept rows keep stable merge keys. The actual fix is the venue-settled exclusion, not a change to how evt_index is assigned.

Given the fix is live and validated, seems safe to close this one out.

**Comment by 0xBoxer** [comment]
issue is fixed indeed
